src/services/EcoBridge/Arbitrum/ArbitrumBridge.selectors.ts

```diff
@@ -0,0 +1,290 @@
+        log: [],
+        bridgeId
+      }
+      if (!tx.partnerTxHash || !l2Txs[tx.partnerTxHash]) {
+        if (tx.type === 'deposit-l1' && tx.receipt?.status !== 0) {
+          summary.status = BridgeTransactionStatus.PENDING // deposits on l1 should never show confirmed on ui
+          summary.pendingReason = ArbitrumPendingReasons.TX_UNCONFIRMED
+        }
+
+        summary.log = createBridgeLog([tx])
+        processedTxsMap[l1ChainId][tx.txHash] = tx.txHash
+
+        total.push(summary)
+        return total
+      }
+
+      if (tx.type === 'outbox') {
+        const status = tx.receipt?.status
+
+        summary.fromChainId = to
+        summary.toChainId = from
+        summary.status = status === 1 ? BridgeTransactionStatus.CLAIMED : getBridgeTxStatus(status)
+        summary.pendingReason = status ? undefined : ArbitrumPendingReasons.TX_UNCONFIRMED
+        summary.log = createBridgeLog([l2Txs[tx.partnerTxHash], tx])
+
+        processedTxsMap[l1ChainId][tx.txHash] = tx.txHash
+        processedTxsMap[l2ChainId][tx.partnerTxHash] = tx.partnerTxHash // skip partner tx in l2Summaries
+
+        total.push(summary)
+        return total
+      }
+
+      if (tx.type === 'deposit-l1' && tx.receipt) {
+        const statusL2 = l2Txs[tx.partnerTxHash].receipt?.status
+        if (tx.receipt?.status === 0 || statusL2 === 0) {
+          summary.status = BridgeTransactionStatus.FAILED
+        } else {
+          summary.status = getBridgeTxStatus(statusL2)
+          summary.pendingReason = statusL2 ? undefined : ArbitrumPendingReasons.DEPOSIT
+          summary.timestampResolved = l2Txs[tx.partnerTxHash].timestampResolved
+        }
+
+        summary.log = createBridgeLog([tx, l2Txs[tx.partnerTxHash]])
+
+        processedTxsMap[l1ChainId][tx.txHash] = tx.txHash
+        processedTxsMap[l2ChainId][tx.partnerTxHash] = tx.partnerTxHash // skip partner tx in l2Summaries
+
+        total.push(summary)
+        return total
+      }
+      return total
+    }, [])
+
+    const l2Summaries = Object.values(l2Txs ?? {}).reduce<BridgeTransactionSummary[]>((total, tx) => {
+      const from = txnTypeToOrigin(tx.type) === 1 ? l1ChainId : l2ChainId
+      const to = from === l1ChainId ? l2ChainId : l1ChainId
+
+      if (processedTxsMap[l2ChainId][tx.txHash]) return total
+
+      const summary: BridgeTransactionSummary = {
+        assetName: tx.assetName,
+        assetAddressL1: tx.assetAddressL1,
+        assetAddressL2: tx.assetAddressL2,
+        value: tx.value,
+        txHash: tx.txHash,
+        batchNumber: tx.batchNumber,
+        batchIndex: tx.batchIndex,
+        fromChainId: from,
+        toChainId: to,
+        status: getBridgeTxStatus(tx.receipt?.status),
+        timestampResolved: tx.timestampResolved,
+        log: [],
+        bridgeId
+      }
+
+      // WITHDRAWAL L2
+      if (!tx.partnerTxHash || !l1Txs[tx.partnerTxHash]) {
+        if (tx.type === 'withdraw') {
+          if (!isLoading) {
```

Comment:
Optionally, since we are doing a simple if/else with this object we might want to invert the condition and the blocks, just to make it easier to read

Reply:
It will be resolved by #895